table/table.go

```diff
@@ -110,6 +113,18 @@ type Table interface {
 	LockRow(ctx context.Context, h int64) error
 }
 
+// UnlimitedScanCnt is used for unlimited scanning.
+const UnlimitedScanCnt = -1
+
+// IsLimit returns if count reachs limit.
+func IsLimit(count, limit int) bool {
```

Comment:
The function seems unnecessary.
Checking the condition in for loop is not hard, e,g. 

for limit != 0 {
   limit--
}

if limit is -1, we will run max int64 time to get 0 again, and we don't have so huge data.